utils/flare.py

```diff
@@ -328,50 +355,56 @@ def _can_read(cls, f, output=True, warn=True):
                 log.warn("  * not readable - {0}".format(f))
             return False
 
-    # Return path to a temp file without comment
-    def _strip_comment(self, file_path):
-        fh, temp_path = tempfile.mkstemp(prefix='dd')
-        atexit.register(os.remove, temp_path)
-        with os.fdopen(fh, 'w') as temp_file:
-            with open(file_path, 'r') as orig_file:
-                for line in orig_file.readlines():
-                    if not self.COMMENT_REGEX.match(line):
-                        temp_file.write(re.sub(self.APIKEY_REGEX, self.REPLACE_APIKEY, line))
-
-        return temp_path
-
-    # Remove password before collecting the file
-    def _add_clean_confd(self, file_path):
+    def _add_clean_conf(self, file_path, target_dir, credential_patterns=None):
         basename = os.path.basename(file_path)
 
-        temp_path, password_found = self._strip_password(file_path)
-        log.info("  * {0}{1}".format(file_path, password_found))
+        temp_path, log_message = self._strip_credentials(file_path, credential_patterns)
+        log.info('  * {0}{1}'.format(file_path, log_message))
         self._add_file_tar(
             temp_path,
-            os.path.join('etc', 'conf.d', basename),
+            os.path.join(target_dir, basename),
             original_file_path=file_path
         )
 
-    # Return path to a temp file without password and comment
-    def _strip_password(self, file_path):
+    # Return path to a temp file without comments on which the credential patterns have been applied
+    def _strip_credentials(self, file_path, credential_patterns=None):
+        if not credential_patterns:
```

Comment:
In Python I prefer avoiding using mutable objects as default parameter values because of this: http://effbot.org/zone/default-values.htm

It can be pretty annoying to debug this kind of behaviors...